src/nidmm/system_tests/test_system_nidmm.py

```diff
@@ -232,29 +234,42 @@ def test_configure_cable_compensation(session):
 
 
 def test_configure_waveform_acquisition(session):
-    session.configure_waveform_acquisition(nidmm.Function.WAVEFORM_VOLTAGE, 100, 100000, 400)
+    session.configure_waveform_acquisition(nidmm.Function.WAVEFORM_VOLTAGE, 100, 1800000, 400)
     assert session.function == nidmm.Function.WAVEFORM_VOLTAGE
     assert session.range == 100
-    assert session.waveform_rate == 100000
+    assert session.waveform_rate == 1800000
     assert session.waveform_points == 400
 
 
 def test_fetch_waveform(session):
-    session.configure_waveform_acquisition(nidmm.Function.WAVEFORM_VOLTAGE, 10, 1000, 10)
+    number_of_points_to_read = 100
+    session.configure_waveform_acquisition(nidmm.Function.WAVEFORM_VOLTAGE, 10, 1800000, number_of_points_to_read)
     with session.initiate():
-        number_of_points_to_read = 10
         measurements, actual_number_of_points = session.fetch_waveform(number_of_points_to_read)
         assert len(measurements) == number_of_points_to_read
         assert isinstance(measurements[1], float)
         assert actual_number_of_points == number_of_points_to_read
 
 
+def test_fetch_waveform_into(session):
+    number_of_points_to_read = 100
+    session.configure_waveform_acquisition(nidmm.Function.WAVEFORM_VOLTAGE, 10, 1800000, number_of_points_to_read)
+    with session.initiate():
+        waveform = numpy.empty(number_of_points_to_read, dtype=numpy.float64)
+        # Initialize with NaN so we can later verify all samples were overwritten by the driver.
+        waveform.fill(float('nan'))
+        measurements = session.fetch_waveform_into(number_of_points_to_read, waveform)
+    for sample in waveform:
+        assert not math.isnan(sample)
+        assert measurements == len(waveform)
```

Comment:
You don't need to do this assert on each sample.

Reply:
Overeager indentation. Fixed.